apps/estuary/lib/estuary/data_writer.ex

```diff
@@ -25,34 +25,48 @@ defmodule Estuary.DataWriter do
   Writes data to PrestoDB and Kafka using `:table_writer` from
   Estuary's application environment.
   """
-  def write(event, opts \\ [])
-
-  def write(%{"author" => _, "create_ts" => _, "data" => _, "type" => _} = event, _) do
-    :ok =
-      event
-      |> make_datawriter_payload()
-      |> @table_writer.write(
-        table: DatasetSchema.table_name(),
-        schema: DatasetSchema.schema()
-      )
+  def write(events, _ \\ []) do
+    payload = make_datawriter_payload(events)
+
+    case get_errors(payload) do
+      [] ->
+        @table_writer.write(payload,
+          table: DatasetSchema.table_name(),
+          schema: DatasetSchema.schema()
+        )
+
+      _ ->
+        {:error, events, "Required field missing"}
+    end
   rescue
-    _ -> {:error, event, "Presto Error"}
+    _ -> {:error, events, "Presto Error"}
   end
 
-  def write(event, _) do
-    {:error, event, "Required field missing"}
+  defp make_datawriter_payload(events) do
+    Enum.map(events, &make_payload/1)
   end
 
-  defp make_datawriter_payload(event) do
-    [
-      %{
-        payload: %{
-          "author" => event["author"],
-          "create_ts" => event["create_ts"],
-          "data" => event["data"],
-          "type" => event["type"]
-        }
+  defp make_payload(%{
```

Comment:
would it make sense to make this a struct and just `new` it up? Might make the pattern match a bit cleaner as well

Reply:
I'd prefer to just use the map here since we're not 're-using' the struct and we'd need to define fields etc. We discussed a bit so I'm going to resolve.